cmd/kpod/create_cli.go

```diff
@@ -40,6 +42,81 @@ func addWarning(warnings []string, msg string) []string {
 	return append(warnings, msg)
 }
 
+func parseVolumes(volumes []string) error {
+	for _, volume := range volumes {
+		arr := strings.Split(volume, ":")
+		switch len(arr) {
+		case 2:
+			if err := validateVolumeHostDir(arr[0]); err != nil {
+				return err
+			}
+			if err := validateVolumeCtrDir(arr[1]); err != nil {
+				return err
+			}
+		case 3:
+			if err := validateVolumeHostDir(arr[0]); err != nil {
+				return err
+			}
+			if err := validateVolumeCtrDir(arr[1]); err != nil {
+				return err
+			}
+			if err := validateVolumeOpts(arr[2]); err != nil {
+				return err
+			}
+		default:
+			return errors.Errorf("incorrect volume format %q, should be host-dir:ctr-dir:[option]", volume)
+		}
+	}
+	return nil
+}
+
+func validateVolumeHostDir(hostDir string) error {
+	if hostDir[0] == '/' {
+		if _, err := os.Stat(hostDir); err != nil {
+			return errors.Wrapf(err, "error checking path %q", hostDir)
+		}
+		return nil
+	}
+
+	// for named volume validation incase we support that in future
+	match, err := regexp.MatchString(`^[\w._\-]+$`, hostDir)
+	if err != nil {
+		return errors.Wrapf(err, "named volume has incorrect format %q", hostDir)
+	}
+	if match {
+		return nil
+	}
+	return errors.Errorf("incorrect format for volume %q", hostDir)
+}
+
+func validateVolumeCtrDir(ctrDir string) error {
+	if ctrDir[0] != '/' {
+		return errors.Errorf("invalid container directory path %q", ctrDir)
+	}
+	return nil
+}
+
+func validateVolumeOpts(option string) error {
+	var foundSharedPrivateSlave, foundRWRO, foundLabelChange int
+	options := strings.Split(option, ",")
+	for _, opt := range options {
+		switch opt {
+		case "rw", "ro":
+			foundRWRO++
```

Comment:
You should check for duplicates.

If founRWRO > 0 {
return errors.Errorf("Can only specify 1 'ro' or 'rw' options")
} 
foundRWRO++

Reply:
fixed